### ISSUE TITLE
Texture format and type check order

### ISSUE THREAD

**Issue by vorg** [regl-project/regl#318]
```javascript
var shadowMap = regl.texture({ width: 1024, height: 1024, format: 'depth', type: 'uint16' })
```

Fails with

`Error: (regl) unknown parameter (uint16): invalid texture type. possible values: uint8,rgba4,rgb565,rgb5 a1 — check.js:20`

Looking at the source code ([texture.js:596](https://github.com/regl-project/regl/blob/gh-pages/lib/texture.js#L596)) it should fail with much more descriptive 

`Error: (regl) you must enable the WEBGL_depth_texture extension in order to use depth/stencil textures. — check.js:20` but it doesn't.

**Comment by Erkaman** [comment]
You're right. That's a pretty bad error message. We should fix that.

**Comment by Erkaman** [comment]
Resolved with PR #320.
